### PR TITLE
Miri subtree sync

### PR DESCRIPTION
**Pull request by oli-obk**
replaces https://github.com/rust-lang/miri/pull/2550

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -275,3 +275,24 @@ see <https://rustc-dev-guide.rust-lang.org/building/how-to-build-and-run.html>.
...
+git remote add miri-your-fork git@github.com:your-name/miri.git
+```
+
+Every time:
+
+```
+ulimit -s 60000
+wget https://raw.githubusercontent.com/gitgitgadget/git/fe2e4819b869725f870cd3ce99f1f8150fe17dc1/contrib/subtree/git-subtree.sh
+sh git-subtree.sh push --prefix src/tools/miri/ miri-your-fork miri-subtree-sync
+rm git-subtree.sh
```
Comment (RalfJung) [inline CONTRIBUTING.md:297]: I think we should just recommend to store that script somewhere. Removing it each time sounds silly?

Comment (oli-obk) [reply CONTRIBUTING.md:297]: XD I thought you wanted to avoid that

Comment (RalfJung) [reply CONTRIBUTING.md:297]: Yeah by storing it in the git repo.^^

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -275,3 +275,24 @@ see <https://rustc-dev-guide.rust-lang.org/building/how-to-build-and-run.html>.
 
 With this, you should now have a working development setup! See
 [above](#building-and-testing-miri) for how to proceed working on Miri.
+
+### Syncing the miri subrree
+
+#### Updating from the rustc repo
```
Comment (RalfJung) [inline CONTRIBUTING.md:281]: What about the other direction?

File: src/bin/miri.rs [hunk 3]
```diff
@@ -7,7 +7,6 @@
 
 extern crate rustc_data_structures;
 extern crate rustc_driver;
-extern crate rustc_errors;
```
[the remark this answers is not in the document]
Comment (oli-obk) [reply src/bin/miri.rs:10]: yea, these are only happening now because rustc has different linting rules for submodules and subtrees

**Pull request closed without merging**